### PR TITLE
BISERVER-12569 - Hex codes added to report title when report is CUT a…

### PR DESCRIPTION
**Pull request by Aliaksandr-Kastenka**
…nd PASTED in BI Console

### CODE REVIEW

**Comment by Aliaksandr-Kastenka** [comment]
@pamval @rmansoor 
Please review.

File: repository/test-src/org/pentaho/platform/repository2/unified/DefaultUnifiedRepositoryBase.java [hunk 1]
```diff
@@ -231,7 +231,7 @@ public void setUp() throws Exception {
     mp.defineInstance( "RepositoryFileProxyFactory", new RepositoryFileProxyFactory( this.jcrTemplate,
         this.repositoryFileDao ) );
     mp.defineInstance( "ITenantedPrincipleNameResolver", new DefaultTenantedPrincipleNameResolver() );
-    mp.defineInstance( "useMultiByteEncoding", Boolean.FALSE );
+    mp.defineInstance( "useMultiByteEncoding", Boolean.TRUE );
     mp.defineInstance( IUnifiedRepository.class, repo );
```
Comment (pamval) [inline repository/test-src/org/pentaho/platform/repository2/unified/DefaultUnifiedRepositoryBase.java]: Why did we need to change this test ? 
Does that mean we changed system behavior  ?

**Comment by Aliaksandr-Kastenka** [comment]
@pamval 
One of the reproduction steps includes setting useMultiByteEncoding=true:

1. Set the following property to <PENTAHO_INSTALLED_DIR>\server\biserver-ee\pentaho-solutions\system\repository.spring.properties
useMultiByteEncoding=false
to
useMultiByteEncoding=true

Class with added test (DefaultUnifiedRepositoryContentIT) extends DefaultUnifiedRepositoryBase
so in order to verify that after fix there is no double encoding this change was made.

**Comment by pamval** [comment]
But aren't you changing the behavior for all tests that use that MicroPlatform ?

**Comment by Aliaksandr-Kastenka** [comment]
Yes, i run all tests and there is no failures.

**Comment by pamval** [comment]
so, what happens if the next guy to work here needs to have that flag set to false again ? 
He changes it and makes your test fail ? 

Can't we just change this flag in the context of this particular test ?

**Comment by Aliaksandr-Kastenka** [comment]
@pamval 
Please review.

**Comment by pamval** [comment]
It appears you just copy pasted the setup method from the base class. Is that so ?

Can I propose the creation of a new method in the base class, initialize(boolean multiByteEncoding) that handles the initialization? And then setup in the base class calls initialize(false) while the setup for this class call initialize(true) ?

**Comment by Aliaksandr-Kastenka** [comment]
@pamval 
Pull-request updated.

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f repository/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  repository/src/org/pentaho/platform/repository2/unified/jcr/JcrRepositoryFileDao.java
  repository/test-src/org/pentaho/platform/repository2/unified/DefaultUnifiedRepositoryBase.java
  repository/test-src/org/pentaho/platform/repository2/unified/DefaultUnifiedRepositoryContentIT.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File repository/src/org/pentaho/platform/repository2/unified/jcr/JcrRepositoryFileDao.java
1. Message: 'if' is not followed by whitespace. at Line: 918 Column: 11
2. Message: 'if' is not followed by whitespace. at Line: 919 Column: 13





Newly Fixed Tests:
-----------------
org.pentaho.platform.repository2.unified.DefaultUnifiedRepositoryAuthorizationIT.testOwnership: org.pentaho.platform.api.repository2.unified.UnifiedRepositoryException
```
org.pentaho.platform.api.repository2.unified.UnifiedRepositoryException: exception while updating ACL for file with id "6a2a504b-f919-4136-913c-094a8618100a"

Reference number: edfe1ae9-ee77-4411-8bda-53ac8d78d5c5
	at org.pentaho.platform.repository2.unified.ExceptionLoggingDecorator.callLogThrow(ExceptionLoggingDecorator.java:512)
	at org.pentaho.platform.repository2.unified.ExceptionLoggingDecorator.updateAcl(ExceptionLoggingDecorator.java:424)
	at org.pentaho.platform.repository2.unified.DefaultUnifiedRepositoryAuthorizationIT.testOwnership(DefaultUnifiedRepositoryAuthorizationIT.java:446)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:74)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:83)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:72)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:231)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:88)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:71)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:174)
Caused by: org.pentaho.platform.api.repository2.unified.UnifiedRepositoryAccessDeniedException: Access denied while updating permissions on file with id [ 6a2a504b-f919-4136-913c-094a8618100a ]
	at org.pentaho.platform.repository2.unified.DefaultUnifiedRepository.updateAcl(DefaultUnifiedRepository.java:435)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
	at org.springframework.security.intercept.method.aopalliance.MethodSecurityInterceptor.invoke(MethodSecurityInterceptor.java:66)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:96)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:260)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:94)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204)
	at com.sun.proxy.$Proxy17.updateAcl(Unknown Source)
	at org.pentaho.platform.repository2.unified.ExceptionLoggingDecorator$39.call(ExceptionLoggingDecorator.java:426)
	at org.pentaho.platform.repository2.unified.ExceptionLoggingDecorator$39.call(ExceptionLoggingDecorator.java:424)
	at org.pentaho.platform.repository2.unified.ExceptionLoggingDecorator.callLogThrow(ExceptionLoggingDecorator.java:489)

```




Integration test coverage change:
============
org.pentaho.platform.repository2.unified.jcr.JcrStringHelper:
 * instructionPercentCoverageChange : +3.33%
 * branchPercentCoverageChange : +5.88%
 * linePercentCoverageChange : +4.00%
 * complexityPercentCoverageChange : +6.67%
    
org.pentaho.platform.repository2.unified.jcr.sejcr.CredentialsStrategySessionFactory:
 * instructionPercentCoverageChange :**-2.57%**
 * branchPercentCoverageChange :**-1.11%**
 * linePercentCoverageChange :**-1.14%**
 * complexityPercentCoverageChange :**-1.18%**

**Pull request merged**